File: src/main/java/me/melijn/melijnbot/enums/Environment.java
package me.melijn.melijnbot.enums;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum Environment {
    PRODUCTION,
    TESTING;


    @JsonCreator
    public static Environment forValue(String value) {
        if ("production".equalsIgnoreCase(value)) {
            return Environment.PRODUCTION;
        } else return Environment.TESTING;
    }
}

File: src/main/java/me/melijn/melijnbot/enums/Environment.kt
package me.melijn.melijnbot.enums

import com.fasterxml.jackson.annotation.JsonCreator

enum class Environment {
    PRODUCTION,
    TESTING;


    companion object {

        @JsonCreator
        fun forValue(value: String): Environment {
            return if ("production" == value.toLowerCase()) {
                Environment.PRODUCTION
            } else Environment.TESTING
        }
    }
}

